src/object/schedcontext.c

```diff
@@ -229,6 +229,15 @@ static exception_t decodeSchedContext_YieldTo(sched_context_t *sc, word_t *buffe
         return EXCEPTION_SYSCALL_ERROR;
     }
 
+    if (NODE_STATE(ksCurThread)->tcbYieldTo != NULL) {
+        // This should not be possible as the currently running thread
```

Comment:
can you also add a assert in case it does happen? Since we can't prove it currently we want to know if we hit it in a test, in a way that fails more obviously than a user error message.

Reply:
Done